From 02b4698deb992876a7fa48f5d46f59d07dde5539 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Tue, 26 Feb 2019 11:58:41 +0200
Subject: [PATCH] Disable SiteSpecificUserAgent.js from the build. Contributes
 to JB#39252

As both "general.useragent.override" and nsISiteSpecificUserAgent
are used and EmbedLite Components provides a UserAgentOverrideHelper
component that implements both complex UA override (register's handler)
and nsISiteSpecificUserAgent that is loaded on startup, let it update
also user agent string that is exposed via DOM's window.navigator.

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 dom/base/Navigator.cpp | 3 +++
 dom/base/moz.build     | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/dom/base/Navigator.cpp b/dom/base/Navigator.cpp
index 290af152b29a..978a98195413 100644
--- a/dom/base/Navigator.cpp
+++ b/dom/base/Navigator.cpp
@@ -1854,6 +1854,8 @@ Navigator::GetUserAgent(nsPIDOMWindowInner* aWindow, nsIURI* aURI,
 {
   MOZ_ASSERT(NS_IsMainThread());
 
+  // Pass call through to nsISiteSpecificUserAgent provided by EmbedLite Components.
+#if 0
   if (!aIsCallerChrome) {
     const nsAdoptingString& override =
       mozilla::Preferences::GetString("general.useragent.override");
@@ -1863,6 +1865,7 @@ Navigator::GetUserAgent(nsPIDOMWindowInner* aWindow, nsIURI* aURI,
       return NS_OK;
     }
   }
+#endif
 
   nsresult rv;
   nsCOMPtr<nsIHttpProtocolHandler>
diff --git a/dom/base/moz.build b/dom/base/moz.build
index 4a4ad7d9349b..3837865fee87 100755
--- a/dom/base/moz.build
+++ b/dom/base/moz.build
@@ -413,7 +413,7 @@ EXTRA_COMPONENTS += [
 ]
 
 # Firefox for Android provides an alternate version of this component
-if CONFIG['MOZ_BUILD_APP'] != 'mobile/android':
+if CONFIG['MOZ_BUILD_APP'] not in ['mobile/android', 'embedding/embedlite']:
     EXTRA_COMPONENTS += [
         'SiteSpecificUserAgent.js',
         'SiteSpecificUserAgent.manifest',
-- 
2.17.1

